src/main/kotlin/com/sourcegraph/cody/ui/AttributionButtonController.kt

```diff
@@ -0,0 +1,55 @@
+package com.sourcegraph.cody.ui
+
+import com.intellij.openapi.project.Project
+import com.intellij.util.concurrency.annotations.RequiresEdt
+import com.sourcegraph.cody.agent.CurrentConfigFeatures
+import com.sourcegraph.cody.agent.protocol.AttributionSearchResponse
+import com.sourcegraph.cody.attribution.AttributionListener
+
+class AttributionButtonController(val button: ConditionalVisibilityButton) : AttributionListener {
+
+  private val extraUpdates: MutableList<Runnable> = ArrayList()
+
+  companion object {
+    fun setup(project: Project): AttributionButtonController {
+      val button = ConditionalVisibilityButton("Attribution search")
+      button.isEnabled = false // non-clickable
+      val currentConfigFeatures: CurrentConfigFeatures =
+          project.getService(CurrentConfigFeatures::class.java)
+      // Only display the button if attribution is enabled.
+      button.visibilityAllowed = currentConfigFeatures.get().attribution
+      return AttributionButtonController(button)
+    }
+  }
+
+  @RequiresEdt
+  override fun onAttributionSearchStart() {
+    button.toolTipText = "Guard Rails: Running Code Attribution Check..."
```

Comment:
nit: Can you please bundle these tool tips texts and texts? 🙏

Reply:
Done!